### PR TITLE
[BugFix] Force _get_batch_preds() to keep logits in decoder timestamp…

### PR DESCRIPTION
**Pull request by tango4j**
…stamps generators

# What does this PR do ?

The argument of `_get_batch_preds` in `FrameBatchASRLogits` class has not been updated when `FrameBatchASR` is updated.
Now, this PR force `_get_batch_preds()` to keep logits in decoder timestamps generators.
This is for r1.18.0 of PR #6499.

**Collection**: [Note which collection this PR will affect]
ASR

# Changelog 

- Added `force_keep_logits` for `_get_batch_preds` function in class `FrameBatchASRLogits` in `decoder_timestamp_utils.py`
- Changed class name `FrameBatchASR_Logits` to `FrameBatchASRLogits`

# Usage
* You can potentially add a usage example below

```python
python offline_diar_with_asr_infer.py \
    diarizer.manifest_filepath=<path to manifest file> \
    diarizer.out_dir='demo_asr_output' \
    diarizer.speaker_embeddings.model_path=<pretrained modelname or path to .nemo> \
    diarizer.asr.model_path=<pretrained modelname or path to .nemo> \
    diarizer.asr.parameters.asr_based_vad=True \
    diarizer.speaker_embeddings.parameters.save_embeddings=False
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

ASR contributors

# Additional Information

This is for main, and there will be another PR for r1.18

### CODE REVIEW

**Review by jbalam-nv — APPROVED** [review]
LGTM
